google-cloud-bigquery/test/google/cloud/bigquery/data_test.rb

```diff
@@ -349,4 +349,125 @@
 
     _(data.class).must_equal Google::Cloud::Bigquery::Data
   end
+
+  describe "statement_type" do
+    let(:data_hash) { { totalRows: nil, rows: [] } }
+
+    it "knows its DDL ALTER_TABLE statement type" do
+      gapi = query_job_resp_gapi "query is ignored", statement_type: "ALTER_TABLE"
+      data = Google::Cloud::Bigquery::Data.from_gapi_json data_hash, nil, gapi, nil
+
+      _(data.statement_type).must_equal "ALTER_TABLE"
+      _(data.ddl?).must_equal true
+      _(data.dml?).must_equal false
+    end
+
+    it "knows its DDL CREATE_MODEL statement type" do
+      gapi = query_job_resp_gapi "query is ignored", statement_type: "CREATE_MODEL"
+      data = Google::Cloud::Bigquery::Data.from_gapi_json data_hash, nil, gapi, nil
+
+      _(data.statement_type).must_equal "CREATE_MODEL"
+      _(data.ddl?).must_equal true
+      _(data.dml?).must_equal false
+    end
+
+    it "knows its DDL CREATE_TABLE statement type" do
+      gapi = query_job_resp_gapi "query is ignored", statement_type: "CREATE_TABLE"
+      data = Google::Cloud::Bigquery::Data.from_gapi_json data_hash, nil, gapi, nil
+
+      _(data.statement_type).must_equal "CREATE_TABLE"
+      _(data.ddl?).must_equal true
+      _(data.dml?).must_equal false
+    end
+
+    it "knows its DDL CREATE_TABLE_AS_SELECT statement type" do
+      gapi = query_job_resp_gapi "query is ignored", statement_type: "CREATE_TABLE_AS_SELECT"
+      data = Google::Cloud::Bigquery::Data.from_gapi_json data_hash, nil, gapi, nil
+
+      _(data.statement_type).must_equal "CREATE_TABLE_AS_SELECT"
+      _(data.ddl?).must_equal true
+      _(data.dml?).must_equal false
+    end
+
+    it "knows its DDL CREATE_VIEW statement type" do
+      gapi = query_job_resp_gapi "query is ignored", statement_type: "CREATE_VIEW"
+      data = Google::Cloud::Bigquery::Data.from_gapi_json data_hash, nil, gapi, nil
+
+      _(data.statement_type).must_equal "CREATE_VIEW"
+      _(data.ddl?).must_equal true
+      _(data.dml?).must_equal false
+    end
+
+    it "knows its DDL \n statement type" do
```

Comment:
Seems odd for a statement type to be the newline character. Are we sure this is correct? (Even if so, I suggest escaping the backslash in this string so there isn't a literal newline in the test name.)

Reply:
It does seem strange but it is the existing behavior--although that behavior was lacking an explicit test!

I'll try to track down the reason for it.

Reply:
Escaped the newline character in the test names.

Reply:
Went and poked at this.  In the broken ALTER TABLE case, `ddlOperationPerformed` is simply missing while `statementType` is set to `ALTER_TABLE`.  I'm not finding an instance of newline values in ddlOperationPerformed or in statementType.

Reply:
@shollyman So can this PR can safely remove the newline character from the `ddl?` [statement types list](https://github.com/googleapis/google-cloud-ruby/blob/8f02c18a3b32025758f1bdfffd23fd9e2077d732/google-cloud-bigquery/lib/google/cloud/bigquery/data.rb#L287)?

Reply:
Yep, approved

Reply:
OK, I'll update the PR to remove the newline character from the list. Thank you!

Reply:
@dazuma I removed the newline character (`\n`) from the `#ddl?` methods' statement types, and their tests.